sdk/core/Azure.Core.TestFramework/src/RecordedTestBaseOfTEnvironment.cs

```diff
@@ -23,5 +27,11 @@ public override void StartTestRecording()
         }
 
         public TEnvironment TestEnvironment { get; }
+
+        [OneTimeSetUp]
+        public async Task WaitForEnvironment()
```

Comment:
Should we make this ValueTask since it will usually complete synchronously?

Reply:
If somebody decides they need "sampling" they'll most likely end up calling API like this:
https://github.com/kasobol-msft/azure-sdk-for-net/blob/dd244f41ee9c999dc09cb42cbd72930f1320770d/sdk/storage/Azure.Storage.Blobs/tests/BlobTestEnvironment.cs#L12-L30
So, this task here would chain to async api call worst case.
Could you elaborate more how would you use ValueTask in this logic?

Reply:
Well, if the method isn't overriden then it would complete synchronously. But I suppose since we are returning Task<bool> the runtime is able to use cached instances for these so it probably doesn't provide much benefit. Also, we'd have to call AsTask anyway when putting them into the dictionary.

https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/#task

Reply:
Good read. How does it look now?
One place I had to retain task is the cache as we shouldn't be await on ValueTasks multiple times (per that article).

Reply:
Yep, that is what I was thinking when I mentioned needing to call AsTask, but I think how you have it is even better.

Reply:
Being that we still have to allocate a task, I don't think there is any practical difference between using `ValueTask<T>` and `Task<T>` here, but ValueTask still seems to convey the intent of the API better. /cc @pakrym to keep me honest here

Reply:
I would say task vs valuetask doesn't matter in this situation at all. We have so many way larger inefficiencies in the test framework.

But `ValueTask<T>` and `Task<T>` difference matters for sync completion mostly. `ValueTask<T>` won't allocate when completed synchronously. But `Task<bool>` return value are cached(by the runtime) so it won't allocate as well